src/Microsoft.Performance.SDK.Runtime.NetCoreApp/Discovery/SandboxPreloadValidator.cs

```diff
@@ -37,15 +38,19 @@ public sealed class SandboxPreloadValidator
 
             this.resolver = new PathAssemblyResolver(allDlls);
             this.versionChecker = versionChecker;
-            this.loadContext = new MetadataLoadContext(this.resolver);
+            this.loadContextsByDirectory = new ConcurrentDictionary<string, MetadataLoadContext>();
         }
 
         public bool IsAssemblyAcceptable(string fullPath, out ErrorInfo error)
         {
+            var loadContext = this.loadContextsByDirectory.GetOrAdd(
+                Path.GetDirectoryName(fullPath),
+                _ => this.CreateLoadContext());
+
             Assembly loaded;
             try
             {
-                loaded = this.loadContext.LoadFromAssemblyPath(fullPath);
```

Comment:
was this intentional

Reply:
Yes, the load context is now retrieved from the dictionary instead of being a single instance variable.

Reply:
Completely missed that :)